hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java

```diff
@@ -303,10 +303,10 @@ public void testThriftExceptionConcurrentCommit() throws TException, Interrupted
 
   @Test
   public void testInvalidObjectException() {
-    TableIdentifier badTi = TableIdentifier.of(DB_NAME, "`tbl`");
+    TableIdentifier badTi = TableIdentifier.of(DB_NAME, "tábl");
```

Comment:
Before Hive 4, the only special character allowed is '/':
https://github.com/apache/hive/blob/branch-3.1/standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java#L155
In Hive 4, all non-alphanumeric characters in a US keyboard, including backtick ('`'), are allowed special characters:
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java#L175-L181

Reply:
Is this test still for hive2?

Reply:
Do we need to test changes before and after hive4?

Reply:
No, we do not. The purpose of the test is simply to test exception handling when the table name is not accepted by Hive. For this purpose, any name not accepted by Hive would do. Previously, a name containing the backtick character ('`') fit the bill, but now we need to use a different kind of name; that's all.